src/Domain/CreateInitialElement.cpp

```diff
@@ -76,18 +79,21 @@ Element<VolumeDim> create_initial_element(
 
     // Refinement in the dimension perpendicular to the interface
     // never causes additional neighbors, even if the neighboring
-    // elements are smaller.
+    // elements are smaller, and the refinement difference is not
+    // restricted to one level.
     {
       auto& perpendicular_segment = gsl::at(segment_ids_of_unrefined_neighbor,
                                             direction_in_neighbor.dimension());
-      auto& perpendicular_refinement =
-          gsl::at(neighbor_is_refined, direction_in_neighbor.dimension());
-      if (perpendicular_refinement) {
+      const size_t neighbor_refinement =
```

Comment:
So I can understand, @wthrowe, could you please explain what this part of the code is doing? I get that you bypass the ERROR() for perpendicular segments, but I don't get what his perpendicular_segment code here does